File: rdos-validation-tools/rdo-validation-tool-java/src/main/java/com/rms/rdo/validation/model/integrityconstraint/IntegrityConstraintValidationError.java
package com.rms.rdo.validation.model.integrityconstraint;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.experimental.FieldDefaults;

@Getter
@Builder
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
public class IntegrityConstraintValidationError {

    String message;

    PrimaryKeyErrorDetail primaryKey;

    ForeignKeyErrorDetail foreignKey;
}

File: rdos-validation-tools/rdo-validation-tool-scala/src/main/scala/com/rms/rdo/validation/model/IntegrityConstraintValidationError.scala
package com.rms.rdo.validation.model

case class IntegrityConstraintValidationError(message: Option[String],
                                              primaryKey: Option[PrimaryKeyErrorDetail],
                                              foreignKey: Option[ForeignKeyErrorDetail])

case class PrimaryKeyError(columnName: String, columnValue: List[String], errorMessage: String)

case class PrimaryKeyErrorDetail(errors: List[PrimaryKeyError])

case class ColumnDetails()

case class ForeignKeyError(file: String, columnValues: Option[List[Map[String, String]]],
                           message: Option[String] = None)

case class ForeignKeyErrorDetail(errorMessage: String, errors: List[ForeignKeyError])